Gems/ROS2/Code/Source/SdfAssetBuilder/SdfAssetBuilder.cpp

```diff
@@ -0,0 +1,199 @@
+        // Once the prefab becomes filled with information, we'll need to do more work here to ensure
+        // that the prefab dependencies are parsed out and hooked up correctly.
+        sdfJobProduct.m_dependenciesHandled = true;
+
+        response.m_outputProducts.push_back(AZStd::move(sdfJobProduct));
+        response.m_resultCode = AssetBuilderSDK::ProcessJobResult_Success;
+    }
+
+    AZStd::string SdfAssetBuilder::CreateDefaultProcPrefab(
+        const AssetBuilderSDK::ProcessJobRequest& request,
+        [[maybe_unused]] AssetBuilderSDK::ProcessJobResponse& response) const
+    {
+        // The following code is similar to DefaultProceduralPrefabGroup::CreatePrefabGroupManifestUpdates().
+        // It might be possible to refactor the two to have shared logic.
+        // It's not worth investigating until the SdfAssetBuilder is fully functional though, because the code
+        // is likely to change pretty drastically still.
+
+        // Create a prefab name based on the source file name.
+        AZStd::string prefabTemplateName{ request.m_sourceFile };
+        AzFramework::StringFunc::Path::ReplaceExtension(prefabTemplateName, "procprefab");
+        AZ::StringFunc::Replace(prefabTemplateName, "\\", "/"); // the source folder uses forward slash
+
+        // clear out any previously created prefab template for this path
+        auto* prefabSystemComponentInterface = AZ::Interface<AzToolsFramework::Prefab::PrefabSystemComponentInterface>::Get();
+        AzToolsFramework::Prefab::TemplateId prefabTemplateId =
+            prefabSystemComponentInterface->GetTemplateIdFromFilePath({ prefabTemplateName.c_str() });
+        if (prefabTemplateId != AzToolsFramework::Prefab::InvalidTemplateId)
+        {
+            prefabSystemComponentInterface->RemoveTemplate(prefabTemplateId);
+            prefabTemplateId = AzToolsFramework::Prefab::InvalidTemplateId;
+        }
+
+        // create a default entity in the prefab to verify functionality.
+        AZ::EntityId entityId;
+        AzToolsFramework::EntityUtilityBus::BroadcastResult(
+            entityId, &AzToolsFramework::EntityUtilityBus::Events::CreateEditorReadyEntity, "Test");
+
+        AZStd::vector<AZ::EntityId> entities = { entityId };
+
+        // create prefab from the "set" of entities (currently just the single default entity)
+        AzToolsFramework::Prefab::PrefabSystemScriptingBus::BroadcastResult(
+            prefabTemplateId,
+            &AzToolsFramework::Prefab::PrefabSystemScriptingBus::Events::CreatePrefabTemplate,
+            entities,
+            prefabTemplateName);
+
+        if (prefabTemplateId == AzToolsFramework::Prefab::InvalidTemplateId)
+        {
+            AZ_Error("prefab", false, "Could not create a prefab template for entities.");
+            return {};
+        }
+
+        // Convert the prefab to a JSON string
+        AZ::Outcome<AZStd::string, void> outcome;
```

Comment:
Not for PR author: Well this is one of the most useless outcomes that exist.
The `PrefabLoaderScriptingBus::SaveTemplateToString` function doesn't return a good value.
Using `void` for the error type negates the entire purpose using an AZ::Outcome.
It would be better if it returned an Error string for the error type and barring that returning an `AZStd::optional<AZStd::string>` to be better express the intent.

Reply:
Yeah, I have no explanation for why the Json/Prefab code uses Outcome<> instead of optional<> for these. My best guess is that it was a stylistic choice to keep API consistency with the other Json/Prefab APIs that use Outcome<>. 

Doing a quick search on Outcome<> in the engine, probably 90% of the uses could be replaced with optional<>. Most of them were probably written before C++ optional<> support was available.